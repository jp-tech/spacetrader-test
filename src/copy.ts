export const COPY = {
    components: {
        buttons: {
            startGame: 'START GAME',
            loading: 'LOADING....',
            accept: 'Accept'
        },
        formsFields: {
            userName: 'Username',
            selectFaction: 'Select Faction'
        },
        statusLabels: {
            accepted: 'Accepted',
            pending: 'Pending',
        },
        tables: {
            contracts: {
                headers: ["CONTRACT TYPE", "AMOUNT", "STATUS", ""],
            },
            userSummary: {
                headers: ["AGENT", "HQ", "FACTION", "SHIPS", "BALANCE" ]
            },
            ships: {
                headers: ["SHIP NAME", "FUEL", "CAPACITY", "SPEED", "CREW", "CONDITION"]
            }
        }
    },
    errors: {
        unexpected: 'An unexpected error occured, please try again'
    },
    pages: {
        startGame: {
            header: 'Welcome to Space Traders ðŸš€',
        },
        notFound: {
            header: 'Not Found :('
        }
    }
}