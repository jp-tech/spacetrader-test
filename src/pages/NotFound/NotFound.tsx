import { COPY } from "../../copy";

export default function NotFound() {
    return (
        <h1>{COPY.pages.notFound.header}</h1>
    );
}
