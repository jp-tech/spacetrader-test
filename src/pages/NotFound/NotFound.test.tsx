import { it, expect } from "vitest";
import { render } from "@testing-library/react";
import NotFound from "./NotFound";
import { COPY } from "../../copy";

it('renders correctly', async () => {
    const {container,getByText} = render(<NotFound />);
    expect(container).toMatchSnapshot();
    expect(getByText(COPY.pages.notFound.header)).toBeInTheDocument();
});
