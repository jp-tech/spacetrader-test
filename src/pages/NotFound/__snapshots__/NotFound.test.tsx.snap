// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<div>
  <h1>
    Not Found :(
  </h1>
</div>
`;
